components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/claims/impl/DefaultClaimHandler.java

```diff
@@ -1100,6 +1107,112 @@ private void handleRoleClaim(AuthenticationContext context, Map<String, String>
         }
     }
 
+    /**
+     * Add the application roles of federated user to remote claims.
+     *
+     * @param stepConfig       StepConfig of current step.
+     * @param context          AuthenticationContext of current authentication flow.
+     * @param idPClaimMappings Claim mappings of IdP of the current step.
+     * @param remoteClaims     Remote claims of the current step authenticated user.
+     * @throws FrameworkException Exception on handling application roles for federated user.
+     */
+    protected void handleApplicationRolesForFederatedUser(StepConfig stepConfig, AuthenticationContext context,
+                                                        ClaimMapping[] idPClaimMappings,
+                                                        Map<String, String> remoteClaims)
+            throws FrameworkException {
+
+        // IdP claim mappings should be available and the current step should be a subject attribute step.
+        if (idPClaimMappings == null || !stepConfig.isSubjectAttributeStep()) {
+            return;
+        }
+        // Get the remote claim URI of the application role claim.
+        String idpAppRoleClaimUri = Arrays.stream(idPClaimMappings)
+                .filter(claimMapping -> claimMapping.getLocalClaim().getClaimUri()
+                        .equals(FrameworkConstants.APP_ROLES_CLAIM))
+                .map(claimMapping -> claimMapping.getRemoteClaim().getClaimUri())
+                .findFirst()
+                .orElse(null);
+        // If there is no application role claim mapping, no need to proceed.
+        if (idpAppRoleClaimUri == null) {
```

Comment:
We can use StringUtils

Reply:
Addressed in 763676b57d4a167a22b90c651e6b0561e0db9d33